### PR TITLE
doc: add arch linux build example

### PR DESCRIPTION
**Pull request by mruddy**
Just adding some notes about building on Arch Linux. Shows which packages are necessary and gives a little info about how the pre-built packages are built.

### CODE REVIEW

**Comment by laanwj** [comment]
Travis failure is unrelated and intermittent (#7463).

BTW for doc-only changes like this you can put `[skip ci]` in the pull request description (not in the title)

**Comment by MarcoFalke** [comment]
> put [skip ci] in the pull request description

I think travis no longer fetches this from GitHub. You'd have to put it in the commit body.

**Comment by MarcoFalke** [comment]
Not sure if it makes sense to hardcode stuff like "v0.12.0" or "python2" as this will be outdated in one or two months.

**Comment by mruddy** [comment]
Cool, I put `[skip ci]` in the commit body and it seems to have worked.
I removed the reference to the tag (tags are around for a long time, but it was not important to what I was documenting so I removed it) and `python2` and reworded things a little.

File: doc/build-unix.md [hunk 1]
```diff
@@ -250,6 +250,24 @@ A list of additional configure flags can be displayed with:
     ./configure --help
 
 
+Setup and Build Example: Arch Linux
+-----------------------------------
+This example lists the steps necessary to setup and build a command line only, non-wallet distribution of the latest changes on Arch Linux:
+
+    pacman -S git base-devel boost libevent python
+    git clone https://github.com/bitcoin/bitcoin.git
+    cd bitcoin/
```
Comment (paveljanik) [inline doc/build-unix.md]: save typing, remove / ;-)

File: doc/build-unix.md [hunk 2]
```diff
@@ -250,6 +250,24 @@ A list of additional configure flags can be displayed with:
     ./configure --help
 
 
+Setup and Build Example: Arch Linux
+-----------------------------------
+This example lists the steps necessary to setup and build a command line only, non-wallet distribution of the latest changes on Arch Linux:
+
+    pacman -S git base-devel boost libevent python
+    git clone https://github.com/bitcoin/bitcoin.git
```
Comment (paveljanik) [inline doc/build-unix.md]: save typing, remove .git at the end

File: doc/build-unix.md [hunk 3]
```diff
@@ -250,6 +250,24 @@ A list of additional configure flags can be displayed with:
...
+    cd bitcoin/
+    ./autogen.sh
+    ./configure --disable-wallet --without-gui --without-miniupnpc
```
Comment (paveljanik) [inline doc/build-unix.md]: Why do you mention miniupnpc here?

**Comment by paveljanik** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/7847/commits/fe53a2af6f8dccd53116ae35f28ec3860cc72f19

**Comment by dcousens** [comment]
concept ACK

File: doc/build-unix.md (hunk 1 above)
Comment (mruddy) [inline doc/build-unix.md]: tab completion :)

File: doc/build-unix.md (hunk 2 above)
Comment (mruddy) [inline doc/build-unix.md]: copy and paste :)

File: doc/build-unix.md (hunk 3 above)
Comment (mruddy) [inline doc/build-unix.md]: didn't need it for what i was doing. this was just from my notes. that's all. i also use `make -j 4 check` (it saves a lot of time to run multiple jobs at once) but didn't want to try to describe how to get the number of non-hyperthreaded cores in a generalized way.

**Comment by laanwj** [comment]
> I think travis no longer fetches this from GitHub. You'd have to put it in the commit body.

Please not in the title next time, though.

**Pull request merged**